ios/REANodesManager.m

```diff
@@ -39,6 +40,12 @@ - (void)setNeedsLayout;
 
 @end
 
+@interface REANodesManager() <RCTUIManagerObserver>
+
+@property BOOL shouldInterceptMountingBlock;
```

Comment:
There is no point in defining this as a property since you only want to use it in the class implementation. In this case you should define it in the block direcly after `@implementation` statement below (e.g., where `_wantRunUpdates` is defined)

Reply:
However, I want to access it by a weak reference.

Reply:
this does not matter, you can always do `weakRef->_shouldInreceptMountikngBlock`

Reply:
Dereferencing a __weak pointer is not allowed due to possible null value caused by race condition, assign it to strong variable first